test/parallel.jl

```diff
@@ -97,6 +97,11 @@ for p in procs(d)
     @test d[idxl] == p
 end
 
+d = SharedArray(Float64, (2,3))
+for x in 1:100000
+    d[:,2]
```

Comment:
This isn't quite a real test, since it doesn't explicitly fail for the reason you want it to.

How about test the return type?